service/storage_service.cc

```diff
@@ -2656,10 +2656,7 @@ int32_t storage_service::get_exception_count() {
     return 0;
 }
 
-future<std::unordered_multimap<dht::token_range, inet_address>> storage_service::get_changed_ranges_for_leaving(sstring keyspace_name, inet_address endpoint) {
-    // FIXME: get erm as param
-    auto& ks = _db.local().find_keyspace(keyspace_name);
-    auto erm = ks.get_effective_replication_map();
+future<std::unordered_multimap<dht::token_range, inet_address>> storage_service::get_changed_ranges_for_leaving(locator::effective_replication_map_ptr erm, inet_address endpoint) {
```

Comment:
Did this also become a pure function? If so it should be static (or free)

Reply:
I think that with `get_ranges_for_endpoint` becoming static this one could potentially become static too.